nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java

```diff
@@ -404,9 +404,26 @@ private Set<ControllerServiceDTO> getControllerServices(final Map<PropertyDescri
     public FlowSnippetDTO copy(final FlowSnippetDTO snippetContents, final ProcessGroup group, final String idGenerationSeed, boolean isCopy) {
         final FlowSnippetDTO snippetCopy = copyContentsForGroup(snippetContents, group.getIdentifier(), null, null, idGenerationSeed, isCopy);
         resolveNameConflicts(snippetCopy, group);
+        removeTopLevelVersionedIds(snippetContents);
```

Comment:
Should this clear the source snippet, as it is doing now, or the copy snippet?

Reply:
Good catch, it was meant to clear the copy since that is what will be used to instantiate the components, will update